odoo/tools/misc.py

```diff
@@ -428,15 +428,33 @@ def mod10r(number):
             report = codec[ (int(digit) + report) % 10 ]
     return result + str((10 - report) % 10)
 
-def str2bool(s, default=None):
-    s = ustr(s).lower()
-    y = 'y yes 1 true t on'.split()
-    n = 'n no 0 false f off'.split()
-    if s not in (y + n):
+
+def str2bool(s: str, default: bool | None = None) -> bool:
+    # allow this (for now?) because it's used for get_param
+    if type(s) is bool:
+        return s
```

Comment:
wasn't there an effort to get rid of `type(x) is y` to favor `isinstance(x, y)` instead?